In [1]:
import pandas as pd
import numpy as np

df=pd.read_csv("../data/raw/deaths_and_causes_synthetic.csv")

df.head()

Year  Country  Gender Age_Group       Cause_of_Death  Number_of_Deaths  \
0  2022    India    Male      0-14    Natural Disasters             73272   
1  2025  Germany  Female      0-14             Homicide            421169   
2  2020    Japan  Female     30-44  Infectious Diseases            103315   
3  2023  Germany    Male     30-44              Suicide            220423   
4  2015  Nigeria  Female     15-29               Stroke            157810   

   Mortality_Rate_per_1000  
0                     2.79  
1                     0.71  
2                     0.75  
3                    11.86  
4                     9.50

In [2]:
df.shape

(260, 7)

In [3]:
df.columns

Index(['Year', 'Country', 'Gender', 'Age_Group', 'Cause_of_Death',
       'Number_of_Deaths', 'Mortality_Rate_per_1000'],
      dtype='object')

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 260 entries, 0 to 259
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Year                     260 non-null    int64  
 1   Country                  260 non-null    object 
 2   Gender                   260 non-null    object 
 3   Age_Group                260 non-null    object 
 4   Cause_of_Death           260 non-null    object 
 5   Number_of_Deaths         260 non-null    int64  
 6   Mortality_Rate_per_1000  260 non-null    float64
dtypes: float64(1), int64(2), object(4)
memory usage: 14.3+ KB


In [5]:
df.describe()

Year  Number_of_Deaths  Mortality_Rate_per_1000
count   260.000000        260.000000               260.000000
mean   2020.126923     257694.530769                 5.956654
std       3.187095     149809.426503                 3.418219
min    2015.000000       4275.000000                 0.190000
25%    2018.000000     113526.500000                 2.912500
50%    2020.000000     265817.000000                 5.955000
75%    2023.000000     391843.000000                 9.040000
max    2025.000000     499331.000000                12.000000

In [6]:
df.nunique()

Year                        11
Country                     10
Gender                       2
Age_Group                    5
Cause_of_Death              11
Number_of_Deaths           260
Mortality_Rate_per_1000    239
dtype: int64

In [7]:
df['Cause_of_Death'].value_counts()

Cause_of_Death
Road Accidents            29
Cardiovascular Disease    29
Infectious Diseases       25
Kidney Disease            25
Cancer                    25
Homicide                  24
Suicide                   23
Respiratory Disease       22
Natural Disasters         20
Stroke                    19
Diabetes                  19
Name: count, dtype: int64

## Step 1 Summary

- The dataset contains 260 records and 7 well-defined columns.
- No missing values were detected.
- Data spans from 2015 to 2025 across 10 countries.
- Causes of death are well balanced, making the dataset suitable for
  exploratory analysis and machine learning tasks.
- The dataset is ready for cleaning, transformation, and deeper analysis
  in the next step.
